azure-toolkit-libs/azure-toolkit-common-lib/src/main/java/com/microsoft/azure/toolkit/lib/common/utils/Utils.java

```diff
@@ -369,4 +371,53 @@ public FileVisitResult visitFileFailed(Path file, IOException exc) {
         }
         return tarFilePath;
     }
+
+    public static boolean isSpringBootProject(Path projectPath) {
+        try {
+            // Check if `pom.xml` exists and contains Spring Boot references
+            Path pomPath = projectPath.resolve("pom.xml");
+            if (Files.exists(pomPath) &&
+                (containsText(pomPath, "org.springframework.boot") ||
+                    containsText(pomPath, "spring-boot-maven-plugin"))) {
+                return true;
+            }
+
+            // Check if `build.gradle` exists and contains Spring Boot references
+            Path gradlePath = projectPath.resolve("build.gradle");
+            if (Files.exists(gradlePath) &&
+                (containsText(gradlePath, "org.springframework.boot") ||
+                    containsText(gradlePath, "spring-boot-starter"))) {
```

Comment:
A probably better way would be checking for plugins `org.springframwork.boot` and `io.spring.dependency-management`.

The same question, should we check for more context rather than short, raw text match?

Reply:
Do we need to check for Gradle case? Given that we are building a Maven plugin?